client/my-sites/pages/controller.js

```diff
@@ -31,9 +30,9 @@ const controller = {
 		context.store.dispatch( setTitle( i18n.translate( 'Site Pages', { textOnly: true } ) ) );
 
 		if ( siteID ) {
-			baseAnalyticsPath = basePath + '/:site';
+			baseAnalyticsPath = status ? `/pages/${ status }/:site` : '/pages/:site';
```

Comment:
@markryall do you maybe know what's preferable here in general case? Should we record `/pages/:status/:site` or `/pages/published/:site` for example. So far we've been recording the latter.

Reply:
So the controller is sometimes too early of a hook point to record page views if it's site specific. (We might not have site in redux state yet). consider using @Copons approach of using the `PageViewTracker` in the Main component.

See example in #23588

Reply:
@gwwar Since I did notice a couple of `/pages/null` and `/pages/undefined` entries in the logs I refactored this to use `PageViewTracker` in 19180df68d035b043f7046afe81a713d614a5a86. However there is a subtle difference in functionality now - previously we were recording a page view for each status change (e.g. `/pages/published/` -> `/pages/drafts`). Now, since the `PageViewTracker` doesn't record page views on re-renders, but just once when it is mounted, these transitions won't be captured. Partly because of that I opted to use the `/pages/:status` format instead. /cc @Copons for thoughts since this might also be a problem for other areas which we might want to refactor to use `PageViewTracker`.

As an alternative, maybe we could enhance `PageViewTracker` with a flag prop that would allow it to record page view events on each re-render?

Reply:
`PageViewTracker` also tracks on receive props, provided `props.path` changes:

https://github.com/Automattic/wp-calypso/blob/master/client/lib/analytics/page-view-tracker/index.jsx#L54

In this case, since both `/published` and `/drafts` are recorded as `/:status`, `PageViewTracker` doesn't record another events as intended.

Unfortunately, "blindly" tracking on each re-render is not a viable option, as we would easily end up accidentally recording the same page view multiple times.

What we might do, though, is to keep track of the actual URL of the page (`window.location.pathname`) instead of `props.path`, and record new page views every time it changes?

Reply:
> In this case, since both /published and /drafts are recorded as /:status, `PageViewTracker` doesn't record another events as intended.

Oh didn't notice that. That's weird, my first version was using exact status name instead of `/:status` but still wasn't logging the changes properly (just the first one). I'll double check this.

> Unfortunately, "blindly" tracking on each re-render is not a viable option, as we would easily end up accidentally recording the same page view multiple times.

Oh I was thinking more in the direction of what we already have (that I've missed) - tracking on `path` prop change. :)

Reply:
Tentative fix for that issue: https://github.com/Automattic/wp-calypso/pull/23685

Reply:
@Copons ok I found out what was the issue - I forgot to rebase this against latest master so I didn't have your latest changes to `PageViewTracker` that included `componentDidUpdate` code :man_facepalming:.

Reply:
@Copons @vindl any quick summary on what folks decided for folks 👀 ? Sounds like this PR doesn't have any blockers otherwise.